Toolkit window stays empty if current cluster isn't reachable

**Steps:**
1. ASSERT: have an unreachable current cluster in current context (ex. stopped minikube)
2. EXEC: launch plugin

**Result:**
The toolkit window stays empty, nothing is shown. Also there's a fatal IDE error:
```java
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:129)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:122)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.restCall(OperationSupport.java:718)
	at io.fabric8.kubernetes.client.impl.BaseClient.getApiGroups(BaseClient.java:292)
	at io.fabric8.kubernetes.client.impl.BaseClient.hasApiGroup(BaseClient.java:208)
	at com.redhat.devtools.intellij.common.kubernetes.ClusterHelper.isOpenShift(ClusterHelper.java:28)
	at com.redhat.devtools.intellij.kubernetes.model.client.ClientAdapter$Factory.create(ClientAdapter.kt:84)
	at com.redhat.devtools.intellij.kubernetes.model.client.ClientAdapter$Factory.create(ClientAdapter.kt:67)
	at com.redhat.devtools.intellij.kubernetes.model.client.ClientAdapter$Factory.create$default(ClientAdapter.kt:59)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$2.invoke(AllContexts.kt:86)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$2.invoke(AllContexts.kt:86)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$client$1.invoke(AllContexts.kt:97)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$client$1.invoke(AllContexts.kt:95)
	at com.redhat.devtools.intellij.kubernetes.model.util.ResettableLazyProperty.get(ResettableLazyProperty.kt:30)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts.getAll(AllContexts.kt:112)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts.getCurrent(AllContexts.kt:103)
	at com.redhat.devtools.intellij.kubernetes.model.ResourceModel.getCurrentContext(ResourceModel.kt:94)
	at com.redhat.devtools.intellij.kubernetes.tree.OpenShiftStructure.canContribute(OpenShiftStructure.kt:184)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeStructure.getValidContributions(TreeStructure.kt:122)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeStructure.createDescriptor(TreeStructure.kt:106)
	at com.intellij.ui.tree.StructureTreeModel$Node.<init>(StructureTreeModel.java:461)
	at com.intellij.ui.tree.StructureTreeModel.getValidRoot(StructureTreeModel.java:397)
	at com.intellij.ui.tree.StructureTreeModel.getRoot(StructureTreeModel.java:314)
	at com.intellij.ui.tree.StructureTreeModel.getRoot(StructureTreeModel.java:37)
	at com.intellij.ui.tree.AsyncTreeModel$CmdGetRoot.computeNode(AsyncTreeModel.java:518)
	at com.intellij.ui.tree.AsyncTreeModel$Command.computeNode(AsyncTreeModel.java:490)
	at com.intellij.util.concurrency.Invoker$Task.run(Invoker.java:381)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:217)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:660)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:735)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:691)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:659)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:202)
	at com.intellij.util.concurrency.Invoker.startTask(Invoker.java:236)
	at com.intellij.util.concurrency.Invoker.invokeSafely(Invoker.java:194)
	at com.intellij.util.concurrency.Invoker.lambda$offerSafely$0(Invoker.java:177)
	at com.intellij.util.concurrency.Invoker$Background.lambda$offer$0(Invoker.java:508)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:244)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:30)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:222)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:218)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:210)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.io.IOException: api.rhamilto.devcluster.openshift.com
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:504)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handleResponse(OperationSupport.java:524)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.restCall(OperationSupport.java:711)
	... 47 more
Caused by: java.net.UnknownHostException: api.rhamilto.devcluster.openshift.com
	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:801)
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306)
	at okhttp3.Dns$1.lookup(Dns.java:40)
	at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:185)
	at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.java:149)
	at okhttp3.internal.connection.RouteSelector.next(RouteSelector.java:84)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:215)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:201)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	... 1 more
```

**Expected result:**
The toolking window should list the existing contexts and the current context should show the error that happened when trying to connect to the cluster.

--- a/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/client/ClientAdapter.kt
+++ b/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/client/ClientAdapter.kt
@@ -81,11 +81,19 @@ abstract class ClientAdapter<C : KubernetesClient>(private val fabric8Client: C)
                     setSslContext(builder, config, externalTrustManagerProvider)
                 }
                 .build()
-            return if (ClusterHelper.isOpenShift(kubeClient)) {
+            return if (isOpenShift(kubeClient)) {
                 val osClient = kubeClient.adapt(NamespacedOpenShiftClient::class.java)
                 OSClientAdapter(osClient, kubeClient)
             } else {
                 KubeClientAdapter(kubeClient)
+            }
+        }
+
+        private fun isOpenShift(client: KubernetesClient): Boolean {
+            return try {
+                ClusterHelper.isOpenShift(client)
+            } catch (e: Exception) {
+                false;
             }
         }
 

--- a/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/util/ExceptionUtils.kt
+++ b/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/util/ExceptionUtils.kt
@@ -11,22 +11,24 @@
 package com.redhat.devtools.intellij.kubernetes.model.util
 
 import io.fabric8.kubernetes.client.Config.DEFAULT_MASTER_URL
+import io.fabric8.kubernetes.client.KubernetesClientException
 import java.net.UnknownHostException
-
-private const val DEFAULT_KUBECLIENT_ERRORMESSAGE = "An error has occurred"
 
 fun toTitle(e: Throwable?): String {
     return noCurrentContextMessage(e)
-        ?: e?.message
-        ?: e?.cause?.message
+        ?: unknownHostMessage(e)
+        ?: unauthorizedMessage(e)
+        ?: extractMessage(e?.message)
+        ?: extractMessage(e?.cause?.message)
         ?: "Error"
 }
 
 fun toMessage(e: Throwable?): String {
     return noCurrentContextMessage(e)
         ?: unknownHostMessage(e)
-        ?: recursiveGetMessage(e)
-        ?: "Unknown Error"
+        ?: unauthorizedMessage(e)
+        ?: extractMessage(recursiveGetMessage(e))
+        ?: "Unknown Error."
 }
 
 private fun recursiveGetMessage(e: Throwable?): String? {
@@ -35,10 +37,38 @@ private fun recursiveGetMessage(e: Throwable?): String? {
     }
     val message = e.message
     if (message == null
-        || message.startsWith(DEFAULT_KUBECLIENT_ERRORMESSAGE)) {
+        || isAnErrorHasOccurred(message)
+        || isOperation(message)
+    ) {
         return recursiveGetMessage(e.cause)
     }
     return message
+}
+
+private fun isOperation(message: String): Boolean {
+    /**
+     * ex. minikube: KubernetesClientException:
+     * "Operation: [list]  for kind: [Service]  with name: [null]  in namespace: [default]  failed."
+     */
+    return message.startsWith("Operation: ")
+            && message.endsWith("failed.")
+}
+
+private fun isAnErrorHasOccurred(message: String) = message.startsWith("An error has occurred")
+
+private fun extractMessage(message: String?): String? {
+    return if (message == null) {
+        null
+    } else if (message.contains("Message: ")) {
+        /**
+         * ex. OpenShift: KubernetesClientException:
+         * "Failure executing: GET at: https://api.sandbox-m3.1530.p1.openshiftapps.com:6443/apis/project.openshift.io/v1/projects.
+         * Message: Unauthorized! Token may have expired! Please log-in again. Unauthorized."
+         */
+        message.substringAfter("Message: ", message)
+    } else {
+        message
+    }
 }
 
 fun noCurrentContextMessage(e: Throwable?): String? {
@@ -50,23 +80,42 @@ fun noCurrentContextMessage(e: Throwable?): String? {
     }
 }
 
-fun unknownHostMessage(e: Throwable?): String? {
-    val unknownHostException = recursiveGetThrowable(e) {
-        throwable -> throwable is UnknownHostException
+private fun unknownHostMessage(e: Throwable?): String? {
+    val unknownHostException = recursiveGetThrowable(e) { throwable ->
+        throwable is UnknownHostException
     }
     return if (unknownHostException is UnknownHostException) {
-        "Unreachable cluster at ${getHost(unknownHostException)}."
+        val host = getHost(unknownHostException)
+        "Unreachable cluster${
+            if (host != null) {
+                " at$host."
+            } else {
+                "."
+            }
+        }"
     } else {
         null
     }
 }
 
 private fun getHost(e: UnknownHostException): String? {
-    val portions = e.message?.split(':') ?: return e.message
+    val portions = e.message?.split(':') ?: return null
     return if (1 < portions.size) {
         portions[1]
     } else {
-        e.message
+        null
+    }
+}
+
+private fun unauthorizedMessage(e: Throwable?): String? {
+    val unauthorizedException = recursiveGetThrowable(e) { throwable ->
+        throwable is KubernetesClientException
+                && throwable.isUnauthorized()
+    }
+    return if (unauthorizedException != null) {
+        "Unauthorized. Verify username and password, refresh token, etc."
+    } else {
+        null
     }
 }
 

--- a/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/util/ExceptionUtils.kt
+++ b/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/util/ExceptionUtils.kt
@@ -24,6 +24,7 @@ fun toTitle(e: Throwable?): String {
 
 fun toMessage(e: Throwable?): String {
     return noCurrentContextMessage(e)
+        ?: unknownHostMessage(e)
         ?: recursiveGetMessage(e)
         ?: "Unknown Error"
 }
@@ -46,5 +47,35 @@ fun noCurrentContextMessage(e: Throwable?): String? {
         "No valid current context in kube config"
     } else {
         null
+    }
+}
+
+fun unknownHostMessage(e: Throwable?): String? {
+    val unknownHostException = recursiveGetThrowable(e) {
+        throwable -> throwable is UnknownHostException
+    }
+    return if (unknownHostException is UnknownHostException) {
+        "Unreachable cluster at ${getHost(unknownHostException)}."
+    } else {
+        null
+    }
+}
+
+private fun getHost(e: UnknownHostException): String? {
+    val portions = e.message?.split(':') ?: return e.message
+    return if (1 < portions.size) {
+        portions[1]
+    } else {
+        e.message
+    }
+}
+
+private fun recursiveGetThrowable(e: Throwable?, predicate: (e: Throwable?) -> Boolean): Throwable? {
+    return if (e == null
+        || e == e.cause
+        || predicate.invoke(e)) {
+        e
+    } else {
+        recursiveGetThrowable(e.cause, predicate)
     }
 }

--- a/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/tree/TreeStructure.kt
+++ b/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/tree/TreeStructure.kt
@@ -28,11 +28,9 @@ import com.redhat.devtools.intellij.kubernetes.model.context.IContext
 import com.redhat.devtools.intellij.kubernetes.model.resource.ResourceKind
 import com.redhat.devtools.intellij.kubernetes.model.util.hasDeletionTimestamp
 import com.redhat.devtools.intellij.kubernetes.model.util.isSameResource
-import com.redhat.devtools.intellij.kubernetes.model.util.isUnauthorized
 import com.redhat.devtools.intellij.kubernetes.model.util.isWillBeDeleted
-import com.redhat.devtools.intellij.kubernetes.model.util.unknownHostMessage
+import com.redhat.devtools.intellij.kubernetes.model.util.toMessage
 import io.fabric8.kubernetes.api.model.HasMetadata
-import io.fabric8.kubernetes.client.KubernetesClientException
 import javax.swing.Icon
 
 /**
@@ -251,49 +249,12 @@ open class TreeStructure(
         project
     ) {
         override fun getLabel(element: java.lang.Exception?): String {
-            return if (isUnauthorized(element)) {
-                "Unauthorized. Verify username and password, refresh token, etc."
-            } else {
-                "Error: ${getMessage(element)}"
-            }
-        }
-
-        private fun isUnauthorized(e: Exception?): Boolean {
-            val cause = e?.cause
-            return cause is KubernetesClientException
-                    && cause.isUnauthorized()
+            return getMessage(element)
         }
 
         private fun getMessage(e: Exception?): String {
-            val unknownHostMessage = unknownHostMessage(e)
-            if (unknownHostMessage != null) {
-                return unknownHostMessage
-            }
-            val causeMessage = e?.cause?.message
-            return if (causeMessage == null) {
-                getMessageOrDefault(e)
-            } else if (causeMessage.contains("Operation: ")) {
-                /**
-                 * ex. minikube: KubernetesClientException:
-                 * "Operation: [list]  for kind: [Service]  with name: [null]  in namespace: [default]  failed."
-                 */
-                return e.cause?.cause?.message
-                    ?: getMessageOrDefault(e)
-            } else {
-                /**
-                 * ex. OpenShift: KubernetesClientException:
-                 * "Failure executing: GET at: https://api.sandbox-m3.1530.p1.openshiftapps.com:6443/apis/project.openshift.io/v1/projects.
-                 * Message: Unauthorized! Token may have expired! Please log-in again. Unauthorized."
-                 */
-                causeMessage.substringAfter("Message: ", causeMessage)
-            }
+            return toMessage(e)
         }
-
-        private fun getMessageOrDefault(e: Exception?): String {
-            return e?.message
-                ?: "unspecified"
-        }
-
 
         override fun getIcon(element: java.lang.Exception): Icon {
             return AllIcons.General.BalloonError

--- a/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/tree/TreeStructure.kt
+++ b/src/main/kotlin/com/redhat/devtools/intellij/kubernetes/tree/TreeStructure.kt
@@ -30,6 +30,7 @@ import com.redhat.devtools.intellij.kubernetes.model.util.hasDeletionTimestamp
 import com.redhat.devtools.intellij.kubernetes.model.util.isSameResource
 import com.redhat.devtools.intellij.kubernetes.model.util.isUnauthorized
 import com.redhat.devtools.intellij.kubernetes.model.util.isWillBeDeleted
+import com.redhat.devtools.intellij.kubernetes.model.util.unknownHostMessage
 import io.fabric8.kubernetes.api.model.HasMetadata
 import io.fabric8.kubernetes.client.KubernetesClientException
 import javax.swing.Icon
@@ -264,6 +265,10 @@ open class TreeStructure(
         }
 
         private fun getMessage(e: Exception?): String {
+            val unknownHostMessage = unknownHostMessage(e)
+            if (unknownHostMessage != null) {
+                return unknownHostMessage
+            }
             val causeMessage = e?.cause?.message
             return if (causeMessage == null) {
                 getMessageOrDefault(e)

--- a/src/test/kotlin/com/redhat/devtools/intellij/kubernetes/model/util/ExceptionUtilsTest.kt
+++ b/src/test/kotlin/com/redhat/devtools/intellij/kubernetes/model/util/ExceptionUtilsTest.kt
@@ -1,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.kubernetes.model.util
+
+import io.fabric8.kubernetes.client.KubernetesClientException
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.UnknownHostException
+
+class ExceptionUtilsTest {
+
+	@Test
+	fun `#getMessage should return 'No valid current context'`() {
+		// given
+		val e = UnknownHostException("https://kubernetes.default.svc")
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).startsWith("No valid current context")
+	}
+
+	@Test
+	fun `#getMessage should return 'Unreachable cluster at'`() {
+		// given
+		val e = UnknownHostException("java.net.UnknownHostException: yoda.io")
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).isEqualTo("Unreachable cluster at yoda.io.")
+	}
+
+	@Test
+	fun `#getMessage should return 'Unreachable cluster' for UnknownHostException`() {
+		// given
+		val e = UnknownHostException("java.net.UnknownHostException")
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).isEqualTo("Unreachable cluster.")
+	}
+
+	@Test
+	fun `#getMessage should return 'Unauthorized verify' for KubernetesClientException with http unauthorized code`() {
+		// given
+		val e = IOException(ResourceException("", KubernetesClientException(null, HttpURLConnection.HTTP_UNAUTHORIZED, null)))
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).startsWith("Unauthorized. Verify")
+	}
+
+	@Test
+	fun `#getMessage should return portion after 'Message' in effective exception message`() {
+		// given
+		val e = IOException(null, ResourceException(null, KubernetesClientException("Failure executing: " +
+				"GET at: https://api.sandbox-m3.1530.p1.openshiftapps.com:6443/apis/project.openshift.io/v1/projects. " +
+				"Message: Unauthorized! Token may have expired! Please log-in again. Unauthorized.")))
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).isEqualTo("Unauthorized! Token may have expired! Please log-in again. Unauthorized.")
+	}
+
+	@Test
+	fun `#getMessage should return cause message in KubernetesClientException with 'Operation'`() {
+		// given
+		val e = IOException(
+			null,
+			ResourceException(
+				null,
+				KubernetesClientException(
+					"Operation: [list]  for kind: [Service]  with name: [null]  in namespace: [default]  failed.",
+					KubernetesClientException(
+						"Failure executing: " +
+								"GET at: https://api.sandbox-m3.1530.p1.openshiftapps.com:6443/apis/project.openshift.io/v1/projects. " +
+								"Message: Unauthorized! Token may have expired! Please log-in again. Unauthorized."
+					)
+				)
+			)
+		)
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).isEqualTo("Unauthorized! Token may have expired! Please log-in again. Unauthorized.")
+	}
+
+
+	@Test
+	fun `#getMessage should return effective exception message`() {
+		// given
+		val e = IOException(null, ResourceException(null, KubernetesClientException("Failure executing: " +
+				"GET at: https://api.sandbox-m3.1530.p1.openshiftapps.com:6443/apis/project.openshift.io/v1/projects. " +
+				"Because cluster is dizzy")))
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).isEqualTo("Failure executing: " +
+				"GET at: https://api.sandbox-m3.1530.p1.openshiftapps.com:6443/apis/project.openshift.io/v1/projects. " +
+				"Because cluster is dizzy")
+	}
+
+	@Test
+	fun `#getMessage should return 'Unknown Error' if no message is present`() {
+		// given
+		val e = IOException(null, ResourceException(null, KubernetesClientException(null, null)))
+		// when
+		val message = toMessage(e)
+		// then
+		assertThat(message).isEqualTo("Unknown Error.")
+	}
+}

